### ISSUE TITLE
No Warning When Using Deprecated Property

### ISSUE THREAD

**Issue by CodeiSir** [palantir/tslint#4763]
### Bug Report

- __TSLint version__: 5.17.0
- __TypeScript version__: 3.5.1
- __Running TSLint via__: Visual Studio

#### TypeScript code being linted

```ts
interface ITest {
  moreStuff: any
  /** @deprecated */
  oldStuff: any
}

const test: ITest = {
  moreStuff: true,
  oldStuff: true,
}
```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint:recommended", "tslint-config-prettier"],
  "rules": {
    "deprecation": true
  }
}
```

#### Actual behavior
does not warn when using deprecated property `oldStuff`

#### Expected behavior
should warn when using deprecated property `oldStuff`

**Comment by JoshuaKGoldberg** [comment]
Ah, this is similar (but not quite the same) as #4682. Accepting PRs - note that it would be a breaking change to start complaining on properties, so this behavior should be enabled by an off-by-default rule option.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
